app/screens/AppNavigator/screens/Dex/PoolSwap/PoolSwapScreen.tsx

```diff
@@ -12,67 +15,125 @@ import { Text, TextInput } from '../../../../../components'
+import { useWallet } from '../../../../../contexts/WalletContext'
+import { useWhaleApiClient } from '../../../../../contexts/WhaleContext'
 import { useTokensAPI } from '../../../../../hooks/wallet/TokensAPI'
 import { translate } from '../../../../../translations'
 import LoadingScreen from '../../../../LoadingNavigator/LoadingScreen'
 import { DexParamList } from '../DexNavigator'
 
+interface DerivedTokenState {
+  id: string
+  amount: string
+  symbol: string
+}
+
 interface SwapSummaryItems {
   poolpair: PoolPairData
-  tokenA: AddressToken
-  tokenB: AddressToken
+  tokenA: DerivedTokenState
+  tokenB: DerivedTokenState
+  tokenAAmount: string
+  tokenBAmount: string
 }
 
 type Props = StackScreenProps<DexParamList, 'PoolSwapScreen'>
 
 export function PoolSwapScreen ({ route }: Props): JSX.Element {
+  const navigation = useNavigation()
   const poolpair = route.params.poolpair
   const tokens = useTokensAPI()
   const [tokenAForm, tokenBForm] = ['tokenA', 'tokenB']
-  const [tokenA, setTokenA] = useState<AddressToken>()
-  const [tokenB, setTokenB] = useState<AddressToken>()
-  const { control, setValue, formState: { isValid }, getValues, trigger } = useForm({ mode: 'onChange' })
+
+  // props derived state
+  const [tokenA, setTokenA] = useState<DerivedTokenState>()
+  const [tokenB, setTokenB] = useState<DerivedTokenState>()
+
+  // component UI state
+  const { control, setValue, formState: { isValid }, getValues, trigger, watch } = useForm({ mode: 'onChange' })
+  const tokenAAmount = watch(tokenAForm)
+  const tokenBAmount = watch(tokenBForm)
+
+  const whaleApiClient = useWhaleApiClient()
+  const account = useWallet().get(0)
 
   function onSubmit (): void {
-    console.log(isValid)
-    console.log(getValues())
+    if (tokenA === undefined || tokenB === undefined) return
+
+    const atA = poolpair.tokenA.id === tokenA?.id ? poolpair.tokenA : poolpair.tokenB
+    const atB = poolpair.tokenA.id === tokenB?.id ? poolpair.tokenA : poolpair.tokenB
+
+    if (tokenA !== undefined && tokenB !== undefined && atA !== undefined && atB !== undefined && isValid) {
+      const swap = {
+        fromToken: tokenA,
+        toToken: tokenB,
+        fromAmount: new BigNumber((getValues()[tokenAForm])),
+        toAmount: new BigNumber((getValues()[tokenBForm]))
+      }
+      // no longer a promise after refactor to network drawer
+      constructSignedSwapAndSend(account, swap, whaleApiClient)
+        .then(() => navigation.dispatch(StackActions.popToTop()))
+        .catch(e => console.log(e))
+    }
   }
 
-  function swapToken (): void {
+  const swapToken = useCallback(async (): Promise<void> => {
     setTokenA(tokenB)
     setTokenB(tokenA)
-  }
+    const { [tokenAForm]: currentA, [tokenBForm]: currentB } = getValues()
+    setValue(tokenAForm, currentB)
+    await trigger(tokenAForm)
+    setValue(tokenBForm, currentA)
+    await trigger(tokenBForm)
```

Comment:
Required to swap the text input values too, else all the form validation messed up.